test/sql/aggregate/group/test_group_by_nested.test

```diff
@@ -0,0 +1,209 @@
+# name: test/sql/aggregate/group/test_group_by_nested.test
+# description: GROUP BY nested types (STRUCTs, LISTs)
+# group: [group]
+
+statement ok
+PRAGMA enable_verification
+
+# Key is LIST<INTEGER>
+statement ok
+CREATE VIEW intlists AS SELECT * FROM (VALUES
```

Comment:
Some more requests for tests:
* Regular comparisons of structs and lists (in a regular WHERE clause or a SELECT clause, i.e. not part of a join)
* Comparison of lists of different types (e.g. LIST<INT> = LIST<VARCHAR> or similar)
* Large lists (e.g. multiple lists that are >vector size)
* IS (NOT) DISTINCT FROM with structs/lists (if this is implemented?)
* Correlated subqueries might also work now (e.g. `select (select a) from (values ([1,2])) tbl(a);`)

Reply:
> Some more requests for tests:
> 
> * Regular comparisons of structs and lists (in a regular WHERE clause or a SELECT clause, i.e. not part of a join)

Comparisons were going to be in the next PR but its not hard to add them now. Although there is a pair of bugs that cancel each other out in the dispatch code(!)

> * Comparison of lists of different types (e.g. LIST = LIST or similar)

This should fail due to a type mismatch?

> * Large lists (e.g. multiple lists that are >vector size)

Do you mean for joins or just predicates?

> * IS (NOT) DISTINCT FROM with structs/lists (if this is implemented?)

I don't believe this is implemented for regular types. At least the switch statement I pulled into `RowOperations::Match` does not have a case for it. But I can add it (and get scalars as well).

> * Correlated subqueries might also work now (e.g. `select (select a) from (values ([1,2])) tbl(a);`)

I'll have a look.

Reply:
> Comparisons were going to be in the next PR but its not hard to add them now. Although there is a pair of bugs that cancel each other out in the dispatch code(!)

Hah. If they are not added yet it is fine to add them in the next PR. Up to you.

> This should fail due to a type mismatch?

At least for normal types we add casts automatically, e.g. `SELECT 1='1'` will add a cast and succeed.

> Do you mean for joins or just predicates?

I would think both.

> I don't believe this is implemented for regular types. At least the switch statement I pulled into RowOperations::Match does not have a case for it. But I can add it (and get scalars as well).

It is implemented in scalar functions (e.g. `select null is not distinct from null` works), which is what I was referring. It is also fine not to have support for that yet in this PR and leave it for a future PR.

Reply:
> > I don't believe this is implemented for regular types. At least the switch statement I pulled into RowOperations::Match does not have a case for it. But I can add it (and get scalars as well).
> 
> It is implemented in scalar functions (e.g. `select null is not distinct from null` works), which is what I was referring. It is also fine not to have support for that yet in this PR and leave it for a future PR.

I'll have to look for what the execution path is because I don't see it! But I had to implement the `DISTINCT` operands for the Postgres comparison semantics, so it should be straightforward to get nested types as well.

Reply:
Think I covered everything, including massive tests for the predicates.